plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/experimental/state/MapboxViewportStateBase.kt

```diff
@@ -0,0 +1,156 @@
+package com.mapbox.maps.plugin.viewport.experimental.state
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.animation.ValueAnimator
+import androidx.core.animation.doOnEnd
+import com.mapbox.maps.plugin.animation.CameraAnimationsPlugin
+import com.mapbox.maps.plugin.viewport.DEFAULT_FRAME_TRANSITION_MAX_DURATION_MS
+import com.mapbox.maps.plugin.viewport.experimental.ViewportPlugin
+import com.mapbox.maps.plugin.viewport.experimental.data.ViewportData
+import com.mapbox.maps.plugin.viewport.experimental.data.ViewportDataSource
+import com.mapbox.maps.plugin.viewport.experimental.data.ViewportDataSourceUpdateObserver
+import com.mapbox.maps.plugin.viewport.experimental.transition.TransitionEndListener
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportStateTransition
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransitionOptions
+import java.util.concurrent.CopyOnWriteArrayList
+
+abstract class MapboxViewportStateBase : ViewportState {
+  protected var frameTransitionOptions =
+    ViewportTransitionOptions.Builder().maxDuration(DEFAULT_FRAME_TRANSITION_MAX_DURATION_MS)
+      .build()
+  protected var runningAnimation: AnimatorSet? = null
+  protected val transitionEndListeners = CopyOnWriteArrayList<TransitionEndListener>()
+
+  abstract val stateTransition: ViewportStateTransition
+  abstract val cameraPlugin: CameraAnimationsPlugin
+
+  abstract val dataSource: ViewportDataSource
+
+  private val dataSourceUpdateObserver =
+    ViewportDataSourceUpdateObserver { viewportData ->
+      if (!isTransitioning) {
+        updateFrame(viewportData, true)
```

Comment:
```suggestion
        updateFrame(viewportData, instant = false)
```
Transition shouldn't be instant by default.

Reply:
It was set to true since the position data coming from LocationComponent is already interpolated values.